pkg/controller/job/job_controller.go

```diff
@@ -1693,11 +1697,32 @@ func findConditionByType(list []batch.JobCondition, cType batch.JobConditionType
 	return nil
 }
 
-func recordJobPodFinished(job *batch.Job, oldCounters batch.JobStatus) {
+func recordJobPodFinished(job *batch.Job, oldCounters batch.JobStatus, newSucceededIndexes []int) {
 	completionMode := completionModeStr(job)
-	diff := job.Status.Succeeded - oldCounters.Succeeded
+	var diff int
+
+	// Updating succeeded metric must be handled differently
+	// for Indexed Jobs to handle the case where the job has
+	// been scaled down by reducing completions & parallelism
+	// in tandem, and now a previously completed index is
+	// now out of range (i.e. index >= spec.Completions).
+	if isIndexedJob(job) {
+		prevUniqueIndexes := parseCompletedIndexes(oldCounters.CompletedIndexes)
+		newUniqueIndexes := uniqueIndexes(newSucceededIndexes)
```

Comment:
Because of this line:

```golang
if ix != unknownCompletionIndex && ix < int(*job.Spec.Completions) && !succeededIndexes.has(ix) {
```

`newSucceededIndexes` should be a disjoint set from `prevUniqueIndexes`. Then all that matter is how many unique indexes there are in `newSucceededIndexes`.